### PR TITLE
BAU: Combine all libraries into two

### PR DESCRIPTION
**Pull request by vixus0**
I'm attempting a change where we combine all the Verify SAML libraries
into two (production and test code). We're already publishing all the individual libraries in
unison so this feels like the next logical step. It was fairly unclear what the
individual responsibilities of each library were in any case.

There are now two libraries, `verify-saml-lib` containing production
code and `verify-saml-test` containing code for use in tests. Hopefully
this makes it easier to remove duplicate code and refactor code in
future.

`verify-saml-lib` contains:
- saml-extensions
- saml-utils
- saml-metadata-bindings
- saml-security
- saml-serializers
- trust-anchor

`verify-saml-test` contains:
- saml-test-utils
- saml-metadata-bindings-test

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -1,8 +1,3 @@
-include "saml-extensions",
-        "saml-security",
-        "saml-utils",
-        "saml-serializers",
-        "saml-metadata-bindings",
-        "saml-metadata-bindings-test",
-        "saml-test-utils",
+include "saml-lib",
+        "saml-test",
         "trust-anchor"
```
Comment (JHjava) [inline settings.gradle]: Can "trust-anchor" be removed too?

Comment (vixus0) [reply settings.gradle]: Oh yeah.

File: saml-lib/build.gradle [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
     compile configurations.opensaml,
-            configurations.dropwizard,
+            configurations.slf4j,
+            configurations.jaxb,
             configurations.guice,
+            configurations.dropwizard,
+            configurations.xml_utils,
             configurations.security,
-            configurations.jaxb,
             project(':trust-anchor')
```
Comment (JHjava) [inline saml-lib/build.gradle]: Can this be nuked too?

Comment (JHjava) [inline saml-lib/build.gradle]:
```suggestion
```

File: saml-lib/build.gradle [hunk 3]
```diff
@@ -1,14 +1,16 @@
 plugins { id "com.jfrog.bintray" version "1.8.4" }
 dependencies {
     compile configurations.opensaml,
-            configurations.dropwizard,
+            configurations.slf4j,
+            configurations.jaxb,
             configurations.guice,
+            configurations.dropwizard,
+            configurations.xml_utils,
             configurations.security,
```
Comment (JHjava) [inline saml-lib/build.gradle]:
```suggestion
            configurations.security
```

**Pull request merged**